### PR TITLE
Added label names for top 10 confidence categories

### PR DESCRIPTION
**Pull request by Tylersuard**
Code will now print out label index, label name, and confidence score for input images.

### CODE REVIEW

**Comment by datumbox** [comment]
Thanks for the contribution.

Ideally we should not need other external libraries to do this. Could you try something like that:
- Keep all ImageNet categories in a list. You can download it the way you do and split on new lines.
- Get from the predictions the top 5 category ids and their probabilities using `torch.topk()`
- Print the category names and scores for the top 5 predictions.

**Comment by Tylersuard** [comment]
Thank you Vasilis.  I just made the requested corrections, let me know if that is what you are looking for.

File: pytorch_vision_resnext.md [hunk 1]
```diff
@@ -63,8 +63,29 @@ with torch.no_grad():
...
 print(output[0])
 # The output has unnormalized scores. To get probabilities, you can run a softmax on it.
-print(torch.nn.functional.softmax(output[0], dim=0))
+probabilities = torch.nn.functional.softmax(output[0], dim=0)
+print(probabilities)
+```
+
+```
+#Download ImageNet labels
+!wget https://raw.githubusercontent.com/Tylersuard/hub/master/imagenet_classes.txt
```
Comment (datumbox) [inline pytorch_vision_resnext.md:72]:
```suggestion
!wget https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt
```

**Comment by datumbox** [comment]
It's looking good Tyler, thanks a lot for the changes. I added one last comment to update the link of the file to point to pytorch's repo.

I'll wait for @pmeier / @fmass 's reply (might take a bit because people started leaving on holidays), to see if there are plans to add the categories directly to TorchVision and avoid downloading them. 

One last non-blocking comment:
image

I assume you left the ID intentionally in case the user intends to link it back to the ImageNet category? We could optionally format a bit the output to indicate clearly which part is the Id, which is the category and which is the probability but I'll leave it to you how you want to have it.

**Comment by pmeier** [comment]
@datumbox 

> Do we currently have any plans to put the ImageNet categories in TorchVision so that we can avoid the ugly/repetitive step 1?

I've already asked about this quite some time ago in https://github.com/pytorch/vision/issues/1647, but as it stands @fmassa was against it for licensing reasons. If this has changed, I'm happy to pick it up and adapt the implementation of `ImageNet` in `torchvision`.

**Comment by datumbox** [comment]
@pmeier Thanks for the reply. I don't know if providing a list with the ImageNet categories has the same issues as redistributing the `meta.bin` file. As far as I can see, other similar libraries from Facebook, Google, Microsoft and other open-source projects ([1](https://github.com/facebookarchive/fb.resnet.torch/blob/master/pretrained/imagenet.lua), [2](https://github.com/tensorflow/tensorflow/search?q=sulphur), [3](https://github.com/microsoft/tensorwatch/blob/master/data/test_images/imagenet_class_index.json), [4](https://github.com/opencv/opencv/blob/b39cd06249213220e802bb64260727711d9fc98c/samples/data/dnn/classification_classes_ILSVRC2012.txt)) include the list of categories in their repos.

@Tylersuard Let me check if there is a blocker on adding the list in the notebook repo and I'll get back to you.

**Comment by pmeier** [comment]
> I don't know if providing a list with the ImageNet categories has the same issues as redistributing the meta.bin file

True, but goes in the same direction. Lets wait for @fmassa s judgement. I'm happy to add them if we want that.

**Comment by Tylersuard** [comment]
Vasilis, thank you.  It does look better without the category number.  I switched back to using the CSV and the index disappears.  I will update the code later today.

**Comment by datumbox** [comment]
@pmeier I just checked internally and we are OK adding the list of categories. I think we should start by adding these here and then discuss options for landing them on TorchVision. What do you think?

@Tylersuard Thanks Tyler. Do you mind if we completely remove the numbers? Since those ids are not related to the ones that ImageNet uses, there is very little reason to keep them. So if you remove them completely from the `.txt` file we should be OK.

**Comment by Tylersuard** [comment]
Ok, I just removed them from the text file.  Please note the download link must have the .raw in the URL or it won't work.

**Comment by datumbox** [comment]
LGTM @Tylersuard. Thanks for the PR and your patience.

To avoid further back and forth and unblock the merge, I pushed directly to your branch a couple of minor changes needed to ensure that the coding style guidelines of PyTorch are met. Please check and let me know if you are OK with them, so that I can merge.

**Review by Tylersuard — COMMENTED** [review]
Looks great!  Much cleaner than my code.

**Comment by Tylersuard** [comment]
Thank you for accepting the changes! Would you like me to add that same change to the other notebooks?

**Comment by datumbox** [comment]
@Tylersuard Yes, please that would be great!

Send a single PR where you apply the same changes that happened on file `pytorch_vision_resnext.md` to the remaining pre-trained models of TorchVision. To keep things simple, please do not update notebooks of other models that are not part of TorchVision, as this might require additional discussions.
